accepted/00000-openapi.md

```diff
@@ -0,0 +1,184 @@
+                schema = @Schema(
+                    type = "object",
+                    properties = {
+                        @StringToClassMapItem(key = "name", value = String.class),
+                        @StringToClassMapItem(key = "description", value = String.class)
+                    }
+                )
+            )
+        )
+	)
+
+```
+
+##### A.1. Library Dependencies
+Two additional Maven dependencies will be added to Uyuni to support Swagger annotations:
+ - `io.swagger.core.v3.swagger-core and`
+ - `io.swagger.core.v3.swagger-annotations`
+
+##### A.2. Existing API Endpoints
+To facilitate the transition, a one-time scanner tool will be developed to automate the conversion of existing Javadoc-based documentation into Swagger annotations. This tool will be implemented as a Java Doclet that systematically scans the codebase to identify predefined patterns in the Javadoc comments. Once identified, the tool will annotate the corresponding methods with the appropriate `@Operation` annotations.
+
+Additionally, the scanner will detect the use of API serializers that describe the return types of endpoints. It will parse the associated Javadoc comments of these serializers to extract the necessary details and incorporate them into the `@Operation` definitions.
+
+##### A.3. XML-RPC API code examples
+
+For XML-RPC endpoints, it is possible to provide a specific section in the web UI containing sample scripts. In addition to this, the `description` and `examples` fields of Swagger operation documentation can be used to aid understanding.
+
+##### A.4. Describing the Request Body
+
+Currently, the request body of an API endpoint is typically mapped to a method parameter of type `java.util.Map<String, Object>`, where the keys represent parameter names. The structure of these parameters is documented in Javadocs using a custom format. For example:
+
+```
+     * @apidoc.param
+ 	*  #struct_begin("props")
+ 	*  	#prop_desc("string", "name", "Content Environment name")
+ 	*  	#prop_desc("string", "description", "Content Environment description")
+ 	*  #struct_end()
+```
+
+To improve the description of the request body, one option is to define a Data Transfer Object (DTO) class and annotate it with Swagger annotations. This approach ensures that an equivalent model is reflected in the API specification.
+
+However, for endpoints where the request body is already defined as a `java.util.Map`, it is possible to define an anonymous model in the specification using the `@StringToClass` annotation. For example:
+
+```
+schema = @Schema(
+    type = "object",
+    properties = {
+        @StringToClassMapItem(key = "label", value = String.class),
+        @StringToClassMapItem(key = "name", value = String.class),
+        @StringToClassMapItem(key = "description", value = String.class)
+    }
+)
+
+```
+
+For migrating the documentation of existing endpoints, this approach—defining the request body as an anonymous model—will be used.
```

Comment:
Agree. It will make the migration simple, and avoid change the method definition, wich can lead to more changes or break the backward compatibility